lib/lambda_ethereum_consensus/validator/duties.ex

```diff
@@ -71,6 +71,9 @@ defmodule LambdaEthereumConsensus.Validator.Duties do
           ValidatorSet.validators()
         ) :: duties()
   def compute_duties_for_epochs(duties_map, epochs_and_start_slots, head_root, validators) do
+    # TODO: This function needs to be measured and optimized if possible, the main point to look
+    # at is the beacon fetch and sync committees computation. Also this could be done asynchronusly
+    # given that except for the first tame it always calculat 1 epoch ahead of time.
```

Comment:
Some information regarding this. The time consumption of sync_committees (which reached 4 seconds on one of my kurtosis tests) is related to the way aggregators are calculated. It has a couple of things very different than attestations which explain why this is costly:
- Attestations are done on one slot per epoch per validator, this is tied to the committee, in sync committees instead, every validator publish in every slot of every epoch during the period of the committee.
- Aggregation is calculated on a per slot basis (we need to compute a signature using the slot) both in attestations and in sync committees, the caveat with sync committees is that they can be aggregators of any slot, which in attestations can only happen once per epoch and here this needs to be checked for the 32 slots.
- This is also calculates for every subcommittee, which in kurtosis main spec every subcommitte is completely repeated due to the low amount of validators (in the test we normally run with less than 120 validators).
- Given all this, for 60 validators (the test I was seeing the 4 seconds delay) we have to calculate proofs for every subcommitte (4) for every slot (32). The proof calculation cost approximately 0.5 ms. so:
  - 60 * 32 * 4 * 0.5 = 3840ms (aprox. 4 seconds)
  
This comment could be removed because:
- If this was an issue, there is not so much room for improvement in the function, but this could be computed asynchronously given that it calculate the next epoch duties ahead of time. This could even be moved some slot further into the new epoch to avoid colliding with the epoch transition.
- In Mainnet this wont be a problem, because having just one validator in a particular period of a sync committee is the norm, more than one in the same period would be statistically improbable. For example, in a test with more participants (10 nodes of 128 validators each and us with 60), the time for computing this was decreased to something near half a second. In a real scenario this will be near a couple of milliseconds.

Reply:
Thanks for the deep dive and the notes. It's pretty clear. I've created the issue #1299 to discuss this for a different PR if needed at all. I suggest adding the issue number to the TODO (all TODOs should refer to an issue IMO).